opaque_keys/__init__.py

```diff
@@ -329,6 +330,47 @@ def replace(self, **kwargs):
         existing_values.update(kwargs)
         return type(self)(**existing_values)
 
+    @classmethod
+    def field_strategy(cls, field):
+        """
+        A strategy to generate data of the right type for the fields of this OpaqueKey.
+
+        Arguments:
+            field: The name of the field to generate data for.
+        """
+        if field == 'deprecated':
+            return strategies.booleans()
+        else:
+            return strategies.text(min_size=1)
+
+    @classmethod
+    def key_strategy(cls):
+        """
+        A strategy to generate instances of this OpaqueKey class.
+        """
+        key_fields = {
+            field: cls.field_strategy(field)
+            for field in cls.KEY_FIELDS
+        }
+        return strategies.builds(
+            cls,
+            deprecated=cls.field_strategy('deprecated'),
+            **key_fields
+        )
+
+    @classmethod
+    def type_strategy(cls, blacklist=None):
+        """
+        A strategy to generate instances of this OpaqueKey KeyType.
+        """
+        if blacklist is None:
+            blacklist = ()
+        return strategies.one_of(*[
+            extension.plugin.key_strategy()
+            for extension in cls._drivers()
+            if not(issubclass(extension.plugin, blacklist))
+        ])
+
```

Comment:
Yeah, that makes sense. We'll need to figure out the best way to differentiate. Sadly, I think we need to keep them on the class so that we can take advantage of subclassing.